src/dex/infusion/infusion.ts

```diff
@@ -0,0 +1,695 @@
+
+    const reserves = await this.getManyPoolReserves(pairsToFetch, blockNumber);
+
+    if (reserves.length !== pairsToFetch.length) {
+      this.logger.error(
+        `Error_getManyPoolReserves didn't get any pool reserves`,
+      );
+    }
+
+    for (let i = 0; i < pairsToFetch.length; i++) {
+      const pairState = reserves[i];
+      const pair = pairsToFetch[i];
+      if (!pair.pool) {
+        await this.addPool(
+          pair,
+          pairState.reserves0,
+          pairState.reserves1,
+          pairState.feeCode,
+          blockNumber,
+        );
+      } else pair.pool.setState(pairState, blockNumber);
+    }
+  }
+
+  async getManyPoolReserves(
+    pairs: InfusionPair[],
+    blockNumber: number,
+  ): Promise<PoolState[]> {
+    try {
+      const multiCallFeeData = pairs.map(pair =>
+        this.getFeesMultiCallData(pair),
+      );
+      const calldata = pairs
+        .map((pair, i) => {
+          let calldata = [
+            {
+              target: pair.token0.address,
+              callData: erc20Iface.encodeFunctionData('balanceOf', [
+                pair.exchange!,
+              ]),
+            },
+            {
+              target: pair.token1.address,
+              callData: erc20Iface.encodeFunctionData('balanceOf', [
+                pair.exchange!,
+              ]),
+            },
+          ];
+          if (this.isDynamicFees) calldata.push(multiCallFeeData[i]!.callEntry);
+          return calldata;
+        })
+        .flat();
+
+      const data: { returnData: any[] } =
+        await this.dexHelper.multiContract.methods
+          .aggregate(calldata)
+          .call({}, blockNumber);
+
+      const returnData = _.chunk(data.returnData, this.isDynamicFees ? 3 : 2);
+
+      return pairs.map((pair, i) => ({
+        reserves0: defaultAbiCoder
+          .decode(['uint256'], returnData[i][0])[0]
+          .toString(),
+        reserves1: defaultAbiCoder
+          .decode(['uint256'], returnData[i][1])[0]
+          .toString(),
+        feeCode: this.isDynamicFees
+          ? multiCallFeeData[i]!.callDecoder(returnData[i][2])
+          : (pair.stable ? this.stableFee : this.volatileFee) || this.feeCode,
+      }));
+    } catch (e) {
+      this.logger.error(
+        `Error_getManyPoolReserves could not get reserves with error:`,
+        e,
+      );
+      return [];
+    }
+  }
+
+  async getSellPrice(
+    priceParams: InfusionPoolOrderedParams,
+    srcAmount: bigint,
+  ): Promise<bigint> {
+    return priceParams.stable
+      ? InfusionStablePool.getSellPrice(priceParams, srcAmount, this.feeFactor)
+      : Uniswapv2ConstantProductPool.getSellPrice(
+          priceParams,
+          srcAmount,
+          this.feeFactor,
+        );
+  }
+
+  async getBuyPrice(
+    priceParams: InfusionPoolOrderedParams,
+    srcAmount: bigint,
+  ): Promise<bigint> {
+    if (priceParams.stable) throw new Error(`Buy not supported`);
+    return Uniswapv2ConstantProductPool.getBuyPrice(
```

Comment:
@JackieJoo I don't understand why we need to return something here if we don't support buy for this integration
<img width="707" alt="BACK-1656: Integrate Infusion by JackieJoo · Pull Request #755 · paraswap:paraswap-dex-lib 🔊 2024-08-23 15-13-00" src="https://github.com/user-attachments/assets/43fb6ac1-a560-4552-93ee-c9bea51b8f06">

Reply:
But it's how we basically signify that a dex doesn't support BUY, you just return null for BUY side pricing, not sure what's the problem?

Reply:
ah, got the point, checking

Reply:
fixed